src/main/scala/cognite/spark/v1/SequenceRowsRelation.scala

```diff
@@ -202,15 +217,24 @@ class SequenceRowsRelation(config: RelationConfig, sequenceId: CogniteId)(val sq
     } else {
       val (columns, fromRowFn) = fromRow(rows.head.schema)
       val projectedRows = rows.map(fromRowFn)
+
       import cats.instances.list._
       projectedRows
-        .groupBy(_.externalId)
+        .groupBy(_.id)
         .toList
         .parTraverse {
           case (cogniteExternalId, rows) =>
-            client.sequenceRows
-              .insertByExternalId(cogniteExternalId.externalId, columns, rows.map(_.sequenceRow))
-              .flatTap(_ => incMetrics(itemsCreated, rows.length))
+            cogniteExternalId match {
+              case externalId: CogniteExternalId =>
+                client.sequenceRows
+                  .insertByExternalId(externalId.externalId, columns, rows.map(_.sequenceRow))
+                  .flatTap(_ => incMetrics(itemsCreated, rows.length))
+              case internalId: CogniteInternalId =>
+                client.sequenceRows
+                  .insertById(internalId.id, columns, rows.map(_.sequenceRow))
+                  .flatTap(_ => incMetrics(itemsCreated, rows.length))
```

Comment:
you don't have to do that now, but it would be nice to migrate these insert functions to CogniteId too ;)